src/v/cluster/types.h

```diff
@@ -501,8 +570,13 @@ struct custom_assignable_topic_configuration {
     operator<<(std::ostream&, const custom_assignable_topic_configuration&);
 };
 
-struct create_partitions_configuration {
+struct create_partitions_configuration
+  : serde::envelope<
+      create_partitions_configuration,
+      serde::version<0>,
+      serde::compat_version<0>> {
```

Comment:
compat_version is optional in this case where we are starting fresh and they are equal.